### DISCUSSION TITLE
Any lightweight, cross-platform desktop application frameworks that perform well when bundled into a single executable?

### DISCUSSION THREAD

**Discussion by bukowa** [nodejs/single-executable#98] [Q&A]
Hi,

I successfully built a single executable following [official instruction](https://nodejs.org/api/single-executable-applications.html). It works fine on Windows (I haven't tested it on other platforms yet). I want my application to be small and lightweight, so I'm looking for a library that will allow me to add a cross-platform desktop UI to my application.

Do you have any recommendations?

**Comment by GabenGar** [comment]
> lightweight
> cross-platform
> UI
> framework

Pick two.

**Comment by bukowa** [comment]
Found this https://github.com/Elanis/web-to-desktop-framework-comparison

**Comment by lb091188** [comment]
Maybe you can try slint-node now~

**Comment by JoCat** [comment]
I'm trying to do something similar by crossing nodejs and webview, but the node sea api is not yet developed enough to make it work properly

**Comment by cspotcode** [comment]
For simple stuff, I've had luck using [deno-windowing/dwm](https://github.com/deno-windowing/dwm) (to open a window) and [deno-windowing/dimgui](https://github.com/deno-windowing/dimgui) (DearIMGUI for the UI)

`deno compile` generates a single executable.

Here's where I posted my experiment:
https://github.com/cspotcode/deno-gui-sandbox

tl;dr is:
```
$ git clone https://github.com/cspotcode/deno-gui-sandbox
$ cd deno-gui-sandbox
$ deno compile --unstable-ffi -A ./imgui2.ts
$ ./imgui2.exe
```
